### PR TITLE
{AKS} Update several test cases related to `aks update`

### PR DESCRIPTION
**Pull request by FumingZhang**
**Description**

Update several test cases related to `aks update`.

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_auto_upgrade_channel.yaml [hunk 1]
```diff
@@ -346,27 +467,27 @@ interactions:
       Connection:
       - keep-alive
       ParameterSetName:
-      - --resource-group --name --location --generate-ssh-keys --enable-managed-identity
-        --auto-upgrade-channel
+      - --resource-group --name --location --enable-managed-identity --auto-upgrade-channel
+        --aks-custom-headers --ssh-key-value
```
Comment (andyliuliming) [inline src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_auto_upgrade_channel.yaml:471]: is aks-custom-headers expected here?

Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_auto_upgrade_channel.yaml:471]: Yes, this is expected.

I have left a comment in the test case, the content specified by the custom header is deprecated and we are only testing the option.

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_auto_upgrade_channel.yaml (hunk 1 above)
Comment (FumingZhang) [reply src/azure-cli/azure/cli/command_modules/acs/tests/latest/recordings/test_aks_create_with_auto_upgrade_channel.yaml]: In commit [#ed6b268](https://github.com/Azure/azure-cli/pull/20463/commits/ed6b26865b11729854154551ceb580f2cf5eddb3), I added a new test case to test the `--aks-custom-headers` option and at the same time this option used in other cases has been removed to ensure consistent testing and user behavior.

**Review by andyliuliming — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**